scale_build/utils/git_utils.py

```diff
@@ -45,3 +45,25 @@ def get_origin_uri(path):
 def push_changes(path, api_token, branch):
     url = urlparse(get_origin_uri(path))
     run(['git', '-C', path, 'push', f'https://{api_token}@{url.hostname}{url.path}', branch])
+
+
+def fetch_origin(path):
+    run(['git', '-C', path, 'fetch', 'origin'])
+
+
+def safe_checkout(path, branch):
+    fetch_origin(path)
+    if branch_exists_in_repository(get_origin_uri(path), branch):
+        run(['git', '-C', path, 'checkout', branch])
+    else:
+        run(['git', '-C', path, 'checkout', '-b', branch])
+
+
+def commit_changes(path, commit_msg):
+    # TODO: Hard coding git username/email for now
+    username = 'bugclerk'
+    email = 'dev@ixsystems.com'
+    run(['git', '-C', path, 'config', 'user.name', username])
+    run(['git', '-C', path, 'config', 'user.email', email])
+    run(['git', '-C', path, 'add', '.'])
+    run(['git', '-C', path, 'commit', '-m', commit_msg])
```

Comment:
Dont need to commit the changes, whoever is doing this will verify first the diff is sane and commit with his creds